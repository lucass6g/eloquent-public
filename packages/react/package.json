{
  "name": "@eloquent/react",
  "version": "0.0.1",
  "description": "",
  "main": "./dist/index.js",
  "files": [
    "dist"
  ],
  "types": "./dist/index.d.ts",
  "exports": {
    ".": {
      "import": "./dist/index.js",
      "require": "./dist/index.js",
      "types": "./dist/index.d.ts"
    }
  },
  "scripts": {
    "build": "tsup"
  },
  "keywords": [],
  "author": "",
  "license": "MIT",
  "devDependencies": {
    "tsup": "^8.0.2",
    "@types/react": "^18.2.66",
    "@types/react-dom": "^18.2.22",
    "react": "^18.2.0",
    "typescript": "^5.4.2"
  }
}
