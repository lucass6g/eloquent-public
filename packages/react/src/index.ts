export {Switch, Case, Default} from './components/conditional-switch'