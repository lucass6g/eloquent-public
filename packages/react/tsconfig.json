{
  "extends": "../../packages/config/base.tsconfig.json",
  "include": ["src"],
  "exclude": ["dist", "build", "node_modules"],
  "compilerOptions": {
    "forceConsistentCasingInFileNames": false,
    "rootDir": "src",
    "baseUrl": ".",
    "paths": {
      "@/*": ["./src/*"]
    }
  }
}
