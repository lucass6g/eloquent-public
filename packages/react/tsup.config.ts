import { defineConfig } from 'tsup';

export default defineConfig({
	name: "@eloquent/react",
	entry: ["src/index.ts"],
	outDir: "dist",
	target: "es2022",
	format: ["cjs", "esm"],
	splitting: false,
	sourcemap: true,
	clean: true,
	dts: true,
	external: ["react", 'react-dom', 'react/jsx-runtime'],
	cjsInterop: true,
	treeshake: true,
})