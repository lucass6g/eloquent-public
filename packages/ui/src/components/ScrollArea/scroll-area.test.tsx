import { render, screen } from '@testing-library/react';
import { describe, it } from 'vitest';
import { ScrollArea } from '.';

describe('ScrollArea Component', () => {
    it('renders horizontal scrollbar', () => {
        const component = render(<ScrollArea.Root data-testid="scroll-area" type="always" className="w-96 whitespace-nowrap rounded-[6px] border">
            <div className="flex w-max space-x-4 p-4">
                O Design System é uma iniciativa para escalar a forma como projetamos e desenvolvemos interfaces e produtos digitais na Elo.
                Todos os componentes foram projetados para serem reutilizados de acordo com a necessidade.
                Nosso objetivo é garantir consistência, acessibilidade e qualidade.
            </div>
            <ScrollArea.ScrollBar data-testid="scrollbar" orientation='horizontal' />
        </ScrollArea.Root>);
        const scrollArea = screen.getByTestId('scroll-area');
        expect(scrollArea).toBeInTheDocument();

        const scrollBar = screen.getByTestId('scrollbar');
        expect(scrollBar).toHaveAttribute('data-orientation', 'horizontal');

        expect(component.baseElement).toMatchSnapshot();
    });

    it('renders vertical scrollbar', () => {
        const component = render(<ScrollArea.Root data-testid="scroll-area" type="always" className="h-[200px] w-[350px] rounded-[6px] border p-4">
            O Design System é uma iniciativa para escalar a forma como projetamos e desenvolvemos interfaces e produtos digitais na Elo.
            Todos os componentes foram projetados para serem reutilizados de acordo com a necessidade.
            Nosso objetivo é garantir consistência, acessibilidade e qualidade.
            <ScrollArea.ScrollBar data-testid="scrollbar" />
        </ScrollArea.Root>);

        const scrollArea = screen.getByTestId('scroll-area');
        expect(scrollArea).toBeInTheDocument();

        const scrollBar = screen.getByTestId('scrollbar');
        expect(scrollBar).toHaveAttribute('data-orientation', 'vertical');

        expect(component.baseElement).toMatchSnapshot();
    });
});