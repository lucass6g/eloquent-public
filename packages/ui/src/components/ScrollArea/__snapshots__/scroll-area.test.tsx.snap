// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ScrollArea Component > renders horizontal scrollbar 1`] = `
<body>
  <div>
    <div
      class="relative overflow-hidden w-96 whitespace-nowrap rounded-[6px] border"
      data-testid="scroll-area"
      dir="ltr"
      style="position: relative; --radix-scroll-area-corner-width: 0px; --radix-scroll-area-corner-height: 0px;"
    >
      <style>
        [data-radix-scroll-area-viewport]{scrollbar-width:none;-ms-overflow-style:none;-webkit-overflow-scrolling:touch;}[data-radix-scroll-area-viewport]::-webkit-scrollbar{display:none}
      </style>
      <div
        class="h-full w-full rounded-[inherit]"
        data-radix-scroll-area-viewport=""
        style="overflow-x: scroll; overflow-y: scroll;"
      >
        <div
          style="min-width: 100%; display: table;"
        >
          <div
            class="flex w-max space-x-4 p-4"
          >
            O Design System é uma iniciativa para escalar a forma como projetamos e desenvolvemos interfaces e produtos digitais na Elo. Todos os componentes foram projetados para serem reutilizados de acordo com a necessidade. Nosso objetivo é garantir consistência, acessibilidade e qualidade.
          </div>
          <div
            class="flex touch-none select-none transition-colors h-2.5 flex-col border-t border-t-transparent p-[1px]"
            data-orientation="horizontal"
            data-testid="scrollbar"
            style="position: absolute; bottom: 0px; left: 0px; --radix-scroll-area-thumb-width: 18px;"
          />
        </div>
      </div>
      <div
        class="flex touch-none select-none transition-colors h-full w-2.5 border-l border-l-transparent p-[1px]"
        data-orientation="vertical"
        style="position: absolute; top: 0px; right: 0px; --radix-scroll-area-thumb-height: 18px;"
      />
    </div>
  </div>
</body>
`;

exports[`ScrollArea Component > renders vertical scrollbar 1`] = `
<body>
  <div>
    <div
      class="relative overflow-hidden h-[200px] w-[350px] rounded-[6px] border p-4"
      data-testid="scroll-area"
      dir="ltr"
      style="position: relative; --radix-scroll-area-corner-width: 0px; --radix-scroll-area-corner-height: 0px;"
    >
      <style>
        [data-radix-scroll-area-viewport]{scrollbar-width:none;-ms-overflow-style:none;-webkit-overflow-scrolling:touch;}[data-radix-scroll-area-viewport]::-webkit-scrollbar{display:none}
      </style>
      <div
        class="h-full w-full rounded-[inherit]"
        data-radix-scroll-area-viewport=""
        style="overflow-x: hidden; overflow-y: scroll;"
      >
        <div
          style="min-width: 100%; display: table;"
        >
          O Design System é uma iniciativa para escalar a forma como projetamos e desenvolvemos interfaces e produtos digitais na Elo. Todos os componentes foram projetados para serem reutilizados de acordo com a necessidade. Nosso objetivo é garantir consistência, acessibilidade e qualidade.
          <div
            class="flex touch-none select-none transition-colors h-full w-2.5 border-l border-l-transparent p-[1px]"
            data-orientation="vertical"
            data-testid="scrollbar"
            style="position: absolute; top: 0px; right: 0px; --radix-scroll-area-thumb-height: 18px;"
          />
        </div>
      </div>
      <div
        class="flex touch-none select-none transition-colors h-full w-2.5 border-l border-l-transparent p-[1px]"
        data-orientation="vertical"
        style="position: absolute; top: 0px; right: 0px; --radix-scroll-area-thumb-height: 18px;"
      />
    </div>
  </div>
</body>
`;
